developing-with-phonon/secure-backend-specification/README.md

```diff
@@ -0,0 +1,279 @@
+
+# Command Glossary
+## LOAD_CA
+This is to load the phonon certificate authority Public Key. Phonon sce devices to be transferred to will have their certificate signatures checked against this key to verify the authenticity of the phonon device.
+### Inputs
+```json
+	{"Certificate":[binary data]}
+```
+
+### Outputs
+```json
+	{}
+```
+
+## INFORMATION
+the phonon sce will return certificate information. Only the identity public key will be returned upon first run, and all the specified information will be returned after LOAD_CA has been run
+### Inputs
+```
+	none
+```
+### Outputs (after certificate loading)
+```json
+	{
+		"pubkey": "binary",
+		"cert": "binary",
+		"name": "string",
+		"nonce": "binary",
+		"counter", "integer or binary?",
+	}
+```
+### Outputs (pre certificate loading)
+```json
+	{
+		"pubkey": "binary",
+	}
+```
+## LOAD_CERT
+The phonon sce will store its identity certificate. the identity certificate will be stored to be returned in the INFORMATION command
+### Inputs
+```json
+	{
+   	"pub": "public key point as byte string",
+   	"iat": "uint32",
+   	"exp": "uint32",
+   	"ver": "int, device firmware version"
+	}
+```
+### Outputs
+```
+	none
+```
+
+## ESTABLISH_SECURITY
+This command is to setup the security mechanism on the phonon sce. This is implementation dependent but SHOULD appropriately lock out unauthorized users from utilizing the phonon sce it is set to. 
+### Inputs
+```json
+	[implementation dependent]
+```
+
+### Outputs:
+```json
+	[implementation dependent]
+```
+## UNLOCK
+This comand is to unlock the phonon sce for use. Before this command is successfully run the phonon sce should not respond to calls to other commands
+### Inputs
+```json
+	[implementation dependent]
+```
+
+### Outputs
+```json
+	[implementation dependent]
+```
+## GENERATE_PHONON
+A phonon keypair will be generated on the phonon sce, and it's public portion will be returned
+### Inputs
+```json
+	{"type":[curve type])
+```
+note: curve type is a string and currently the only option is "secp256k1"
+### Outputs
+```json
+	{"pubkey":"binary data"}
+```
+## SET_DESCRIPTOR
+the descriptor of a phonon contains arbitrary key value pairs and is to be attached to phonons when this command is run. the identifier of the phonon is implementation dependent and may be a representation of the pubkey or an index or other identifier stored on the sce.
+### Inputs
+```json
+	{
+		{
+			phonon:4
+		}
+		"data":{
+		"arbitrary keys":"arbitrary values"
+		}
+	}
+```
+
+### Outputs
+	```json
+	```
+## LIST_PHONONS
+### Inputs
+	```json
+	[optional]
+	```json
+### Outputs
+```json
+	[
+	{
+		"header": {
+			"curve": "secp256k1",
+			"pub": "public key bytes"
+			},
+		"metadata":{
+			"xyz": "abc",
+			"something else":"another value",
+		}
+	]
+```
+note: some implemntations will allow for filter queries to be input to cut down on the amount of phonons returned in a single request
```

Comment:
There will be no filters, as the metadata is opaque to the applet. Also, not all phonons will be returned "at once". one will need to "loop-list" them.